doc/api/process.markdown

```diff
@@ -59,8 +59,9 @@ finished running the process will exit. Therefore you **must** only perform
 checks on the module's state (like for unit tests). The callback takes one
 argument, the code the process is exiting with.
 
-This event may not be fired if the process terminates due to signals like
-`SIGINT`, `SIGTERM`, `SIGKILL`, and `SIGHUP`.
+This event is only emitted when node exits explicitly by process.exit() or
+implicitly by the event loop draining. It will not, for example, be fired
```

Comment:
I don't think it's as easy to identify every signal that can cause this as previous conversation may have made it seem. Previous conversation/links suggested five:  `SIGINT`, `SIGTERM`, `SIGQUIT`, `SIGKILL`, and `SIGHUP`. But there's also `SIGILL`, `SIGABRT`, and probably others. And there's non-POSIX signals that are supported on common OSes, like `SIGPWR` which is supported by Linux (which I didn't test, so I don't know if it manifests in this behavior, but that's kind of my point).

You might be able to summarize those signals without itemizing them. Something like "a signal that causes program termination" might suffice. But if the goal was to be comprehensive, we also need to describe errors that causes `exit` firing to not happen. (`SyntaxError`? No `exit` event. `ReferenceError`? Then you get an `exit` event.) Also: uncaught exceptions. Possibly other things I'm not thinking of too.

It may be possible to generalize these things. But that risks ending up just repeating the previous sentence without shedding any light. In that regard, I think a common example is useful.

I actually don't feel nearly as strongly about this as this comment makes it sound. But I think if we're going to do something different, we should definitely take all this into consideration.